cl/donate/api_views.py

```diff
@@ -0,0 +1,215 @@
+    mixins.CreateModelMixin, viewsets.GenericViewSet
+):
+    permission_classes = [AllowNeonWebhook]
+    serializer_class = NeonMembershipWebhookSerializer
+    queryset = NeonMembership.objects.all()
+
+    def create(self, request: Request, *args, **kwargs):
+        """
+        Processes membership webhooks received from Neon CRM.
+
+        This method handles POST requests containing JSON data from Neon CRM.
+        It extracts relevant information from the data and updates the
+        corresponding membership record in the database.
+
+        Args:
+            request (Request): A request object containing the webhook data.
+
+        Returns:
+            Response: A HttpResponse object with status code 200 if successful,
+            or another appropriate response if an error occurs.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        webhook_data = serializer.validated_data
+        self._store_webhook_payload(webhook_data)
+        match webhook_data["eventTrigger"]:
+            case "createMembership" | "editMembership":
+                self._handle_membership_creation_or_update(webhook_data)
+            case "deleteMembership":
+                self._handle_membership_deletion(webhook_data)
+            case "updateMembership":
+                pass
+            case _:
+                trigger = webhook_data["eventTrigger"]
+                raise NotImplementedError(f"Unknown event trigger-{trigger}")
+
+        return Response(status=HTTPStatus.CREATED)
+
+    def _get_member_record(self, account_id: str) -> User:
+        """
+        Retrieves a user record associated with a Neon CRM account ID.
+
+        This method attempts to find a matching Django user in two ways:
+
+        1. first tries to directly query the database for a user whose
+        `neon_account_id` field matches the provided `account_id`.
+        2. If no matching user is found in the database, it fetches the
+        account email address from the Neon API using the `account_id`.
+        It then tries to find a user whose email address matches the
+        retrieved Neon account email.
+
+        Args:
+            account_id (str): Unique identifier assigned by Neon to an account
+
+        Returns:
+            User: User object associated with the Neon account
+        """
+        try:
+            user = User.objects.get(profile__neon_account_id=account_id)
+        except User.DoesNotExist:
+            client = NeonClient()
+            neon_account = client.get_acount_by_id(account_id)
+            users = User.objects.filter(
+                email=neon_account["primaryContact"]["email1"]
+            )
+            if not users:
+                return HttpResponse(
+                    "Error processing webhook, User not found",
+                    status=HTTPStatus.INTERNAL_SERVER_ERROR,
+                )
+            user = users.first()
```

Comment:
User objects don't have a default ordering that I'm aware of, and you didn't apply ordering in your query up above, so this call to `first()` is undefined in how it behaves. That's no good. 

On top of that, I actually wonder what we *should* do here. People are currently allowed to make more than one account in CL with the same email address, and indeed people *do*. I guess we should ban that eventually, but maybe the best thing to do here is order by most recently logged in, and update that account?

Reply:
> maybe the best thing to do here is order by most recently logged in, and update that account?

I believe that this is a good idea. I'll update the code.